wetterdienst/dwd/radar/index.py

```diff
@@ -0,0 +1,221 @@
+def create_fileindex_radolan_grid(time_resolution: TimeResolution) -> pd.DataFrame:
+    """
+    Function used to create a file index for the RADOLAN product. The file index will
+    include both recent as well as historical files. A datetime column is created from
+    the filenames which contain some datetime formats. This datetime column is required
+    for later filtering for the requested file.
+
+    Args:
+        time_resolution: time resolution enumeration for the requested RADOLAN product
+        where two are possible: hourly and daily
+
+    Returns:
+        file index as DataFrame
+    """
+    file_index = pd.concat(
+        [
+            create_fileindex_radar(
+                parameter=RadarParameter.RADOLAN_GRID,
+                time_resolution=time_resolution,
+                period_type=period_type,
+            )
+            for period_type in (PeriodType.HISTORICAL, PeriodType.RECENT)
+        ]
+    )
+
+    file_index = file_index[
+        file_index[DWDMetaColumns.FILENAME.value].str.contains("/bin/")
+        & file_index[DWDMetaColumns.FILENAME.value].str.endswith(
+            (ArchiveFormat.GZ.value, ArchiveFormat.TAR_GZ.value)
+        )
+    ]
+
+    # Decode datetime of file for filtering.
+    pattern = re.compile(f"{RADOLAN_HISTORICAL_DT_REGEX}|{RADOLAN_RECENT_DT_REGEX}")
+    file_index[DWDMetaColumns.DATETIME.value] = file_index[
+        DWDMetaColumns.FILENAME.value
+    ].apply(
+        lambda filename: parse(
+            pattern.findall(filename)[0],
+            date_formats=[DatetimeFormat.YM.value, DatetimeFormat.ymdhm.value],
+        )
+    )
+
+    return file_index
+
+
+def build_path_to_parameter(
+    parameter: RadarParameter,
+    time_resolution: Optional[TimeResolution] = None,
+    period_type: Optional[PeriodType] = None,
+    radar_site: Optional[RadarSite] = None,
+    radar_data_type: Optional[RadarDataType] = None,
+) -> str:
+    """
+    Function to build a indexing file path.
+    Supports composite- and site-based radar data.
+
+    Composites
+    ----------
+    - https://opendata.dwd.de/climate_environment/CDC/grids_germany/5_minutes/radolan/
+    - https://opendata.dwd.de/climate_environment/CDC/grids_germany/daily/radolan/
+    - https://opendata.dwd.de/climate_environment/CDC/grids_germany/hourly/radolan/
+    - https://opendata.dwd.de/weather/radar/composit/
+    - https://opendata.dwd.de/weather/radar/radolan/
+
+    Sites
+    -----
+    - https://opendata.dwd.de/weather/radar/sites/
+
+
+    Args:
+        parameter: observation measure
+        time_resolution: frequency/granularity of measurement interval
+        period_type: recent or historical files
+        radar_site: Site of the radar if parameter is one of RADAR_PARAMETERS_SITES
+        radar_data_type: Some radar data are available in different data types
+
+    Returns:
+        indexing file path relative to climate observations path
+    """
+    if parameter == RadarParameter.RADOLAN_GRID:
```

Comment:
Can we exchange this if/else conditionals to a dict based mapping?
E.g.:
PARAMETER_PATH_MAPPING[parameter].format(time_resolution=time_resolution.value, period_type=period_type.value)

Note: .format allows you to add more parameter than given in the string. That makes it a very nice feature in this context

Reply:
While I am not totally against this, I would like to keep it that way for now until we have resolved the HDF5 untangling problem outlined below. I believe there are too many conditions and some of them might require manual adjustments, so that it will not be that easy to gain improved conciseness from introducing another mapping here.

> Note: `.format()` allows you to add more parameter than given in the string. That makes it a very nice feature in this context.

While I believe I know what you are pointing out here, please let me also point out that it is not so easy to account for putting the path separator `/` at the appropriate places when the path to be computed has more fragments than others.

Please consider this snippet re. `RADAR_PARAMETERS_SITES` as an example:
```
parameter_path = f"weather/radar/sites/{parameter.value}/{radar_site.value}"
if parameter in RADAR_PARAMETERS_WITH_HDF5:
    if radar_data_type is None:
        raise ValueError("RadarDataType missing [hdf5 or binary]")
    elif radar_data_type is RadarDataType.HDF5:
        parameter_path = f"{parameter_path}/{radar_data_type.value}"
```

When using `RadarDataType.HDF5` here, the path will have to be extended by another fragment and this fragment has to be separated by another path seperator `/`. Folding this into the graceful behavior we might gain from `.format()` you have pointed out will not be sufficient.

I believe the function `build_path_to_parameter()` is straightforward right now and will easily leave room for adjusting to all edge cases we are still trying to tackle. I believe introducing another mapping here will get us more in the way of implementing all these edge cases in a clean and concise manner.